src/components/views/elements/AccessibleTooltipButton.tsx

```diff
@@ -29,71 +29,56 @@ interface IProps extends React.ComponentProps<typeof AccessibleButton> {
     alignment?: Alignment;
     onHover?: (hovering: boolean) => void;
     onHideTooltip?(ev: SyntheticEvent): void;
-}
+};
 
-interface IState {
-    hover: boolean;
-}
+const AccessibleTooltipButton = forwardRef(function <T extends keyof JSX.IntrinsicElements>(
+    { title, tooltip, children, forceHide, alignment, onHideTooltip, tooltipClassName, ...props }: Props<T>,
+    ref: Ref<HTMLElement>,
+) {
+    const [hover, setHover] = useState(false);
 
-export default class AccessibleTooltipButton extends React.PureComponent<IProps, IState> {
-    public constructor(props: IProps) {
-        super(props);
-        this.state = {
-            hover: false,
-        };
-    }
-
-    public componentDidUpdate(prevProps: Readonly<IProps>): void {
-        if (!prevProps.forceHide && this.props.forceHide && this.state.hover) {
-            this.setState({
-                hover: false,
-            });
+    useEffect(() => {
+        if (forceHide && hover) {
```

Comment:
it was clear what this was doing before, but that clarity has been lost in translation.
```suggestion
        // If `forceHide` is now set, we need to clear the `hover` state if it was set (since you
        // can't hover over a hidden element).
        if (forceHide && hover) {
```

Reply:
> can't hover over a hidden element

The hide isn't related to the element but the tooltip, I thought that this in combination with the added doc on the `forceHide` prop was sufficient personally

Reply:
> The hide isn't related to the element but the tooltip, I thought that this in combination with the added doc on the forceHide prop was sufficient personally

The fact I've misunderstood what's going on here is a strong indication it needs a better comment, tbh.

Why *are* we forcing `hover` to off? Is it because that's what actually hides the tooltip?

Reply:
I believe so, though this code wasn't written in this PR, just refactored into a Functional Component